master/buildbot/buildslave/base.py

```diff
@@ -201,55 +209,57 @@ def setServiceParent(self, parent):
         # botmaster needs to set before setServiceParent which calls startService
 
         self.manager = parent
-        self.master = parent.master
-        self.botmaster = parent.master.botmaster
-        return service.AsyncMultiService.setServiceParent(self, parent)
+        return service.BuildbotService.setServiceParent(self, parent)
 
     @defer.inlineCallbacks
     def startService(self):
         self.updateLocks()
         self.startMissingTimer()
-
         self.buildslaveid = yield self.master.data.updates.findBuildslaveId(
-            self.slavename)
+            self.name)
 
         yield self._getSlaveInfo()
-        yield service.AsyncMultiService.startService(self)
+        yield service.BuildbotService.startService(self)
 
     @defer.inlineCallbacks
-    def reconfigServiceWithBuildbotConfig(self, new_config):
-        # Given a new BuildSlave, configure this one identically.  Because
-        # BuildSlave objects are remotely referenced, we can't replace them
+    def reconfigService(self, name, password, max_builds=None,
+                        notify_on_missing=None, missing_timeout=3600,
+                        properties=None, locks=None, keepalive_interval=3600):
+        # Given a BuildSlave config arguments, configure this one identically.
+        # Because BuildSlave objects are remotely referenced, we can't replace them
         # without disconnecting the slave, yet there's no reason to do that.
-        new = self.findNewSlaveInstance(new_config)
-
-        assert self.slavename == new.slavename
-        self.password = new.password
 
-        # update our records with the buildslave manager
-        if not self.registration:
-            self.registration = yield self.master.buildslaves.register(self)
-        yield self.registration.update(new, new_config)
+        assert self.name == name
+        self.password = password
 
         # adopt new instance's configuration parameters
-        self.max_builds = new.max_builds
-        self.access = new.access
-        self.notify_on_missing = new.notify_on_missing
+        self.max_builds = max_builds
+        self.access = []
+        if locks:
+            self.access = locks
+        self.notify_on_missing = notify_on_missing
 
-        if self.missing_timeout != new.missing_timeout:
+        if self.missing_timeout != missing_timeout:
             running_missing_timer = self.missing_timer
             self.stopMissingTimer()
-            self.missing_timeout = new.missing_timeout
+            self.missing_timeout = missing_timeout
             if running_missing_timer:
                 self.startMissingTimer()
 
-        properties = Properties()
-        properties.updateFromProperties(new.properties)
-        self.properties = properties
+        if properties is None:
```

Comment:
it seems 'properties'  is not used anymore, as everything is computed in self.properties